tests/clone/nonetwork.c

```diff
@@ -353,44 +353,19 @@ void test_clone_nonetwork__clone_from_empty_sets_upstream(void)
 	cl_fixture_cleanup("./repowithunborn");
 }
 
-static int just_return_origin(git_remote **out, git_repository *repo, const char *name, const char *url, void *payload)
-{
-	GIT_UNUSED(url); GIT_UNUSED(payload);
-
-	return git_remote_lookup(out, repo, name);
-}
-
-static int just_return_repo(git_repository **out, const char *path, int bare, void *payload)
-{
-	git_submodule *sm = payload;
-
-	GIT_UNUSED(path); GIT_UNUSED(bare);
-
-	return git_submodule_open(out, sm);
-}
-
 void test_clone_nonetwork__clone_submodule(void)
 {
-	git_clone_options clone_opts = GIT_CLONE_OPTIONS_INIT;
 	git_index *index;
 	git_oid tree_id, commit_id;
 	git_submodule *sm;
 	git_signature *sig;
-	git_repository *sm_repo;
 
 	cl_git_pass(git_repository_init(&g_repo, "willaddsubmodule", false));
 
-
 	/* Create the submodule structure, clone into it and finalize */
 	cl_git_pass(git_submodule_add_setup(&sm, g_repo, cl_fixture("testrepo.git"), "testrepo", true));
-
-	clone_opts.repository_cb = just_return_repo;
-	clone_opts.repository_cb_payload = sm;
-	clone_opts.remote_cb = just_return_origin;
-	clone_opts.remote_cb_payload = sm;
-	cl_git_pass(git_clone(&sm_repo, cl_fixture("testrepo.git"), "testrepo", &clone_opts));
+	cl_git_pass(git_submodule_add_clone(NULL, sm));
```

Comment:
I think we should have a new test that uses `git_submodule_add_clone`, as we should retain the test that verifies the old way of doing that.